src/PointwiseFunctions/AnalyticSolutions/Errors.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Tags {
+/*!
+ * \brief Prefix indicating the difference between two values, such as betwen a
+ * numeric quantity and the corresponding quantity computed from an analytic
+ * solution.
+ */
+template <typename Tag>
+struct Difference : db::PrefixTag, db::SimpleTag {
+  using type = db::const_item_type<Tag>;
+  using tag = Tag;
+};
+
+/*!
+ * \brief For each `Tag` in `TagsList`, where `TagsList` is a subset of
+ * `AnalyticTagsList`, compute the difference `Tag` - `Tags::Analytic<Tag>`.
+ */
+template <typename TagsList, typename AnalyticTagsList>
+struct DifferenceFromAnalyticCompute
+    : db::add_tag_prefix<::Tags::Difference, ::Tags::Variables<TagsList>>,
+      db::ComputeTag {
+  using argument_tags = tmpl::list<
+      Tags::Variables<TagsList>,
+      db::add_tag_prefix<Tags::Analytic, Tags::Variables<AnalyticTagsList>>>;
```

Comment:
why isn't this simply `Tags::Variables<AnalyticTagsList>` ?

Reply:
The analytic Variables is keyed by Tags::Analytic<Tag> , while the numeric Variables is keyed by Tag. I couldn't figure out how to do what you suggest, because then how do I figure out which Tag goes with which AnalyticTag? But if there's a better way, Id welcome the suggestion!

Reply:
where does `AnalyticTagsList` come from?  Why isn't it `tmpl::list<Tags::Analytic<Var0>,...Tags::Analytic<VarN>>` ?

Reply:
@kidder For variables tags we need the `add_tag_prefix` because both the variables tag as well as its containing tags must be prefixed to work correctly.

And I believe the list of tags we have an analytic solution for can be a subset of the full list of system variables, so I think it makes sense to have the separate template parameter. It could default to the full tags list though.

How about you replace the `TagsList` template parameter with `VariablesTag`? You can get its tags list as `db::get_variables_tags_list<VariablesTag>` so you have the same information available, but you don't have to build the `::Tags::Variables`. If the tags in `TagsList` carry prefixes then the code you currently have would not compile.

Reply:
After working on this some more, I figured out what's going on here.

Yes, AnalyticTagsList is `temple::list<Tags::Analytic<Varo>, ... Tags::Analytic<VarN>>`.

The trick I didn't see before was that it should actually be

`Tags::Analytic<Tags::Variables<AnalyticTagsList>>`, because not only are the tags in the `Variables` wrapped in `Tags::Analytic`, but the `Variables` itself is as well.

This change comes at the price that users that call `DifferenceFromAnalyticCompute` must themselves wrap the list of tags provided by the analytic solution in Tags::Analytic<>, because the solutions' `tags` aliases don't do this.

I figure you would be ok with that price, so I made this change as you requested and updated the test that calls it accordingly.